From 0365891dd5f2e0c7f6ca638941c1f96e593e9cb9 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Thu, 18 Mar 2021 17:03:02 -0700
Subject: [PATCH 1788/1852] drm: xlnx: zynqmp_disp: Release VTC bridge in
 unbind stage

commit d58603c63e921b11448b8abcccc226969613916d from
https://github.com/Xilinx/linux-xlnx.git

Currently, when VTC platform device is unregistered, it is still owned
by DP from bridge driver's point of view. This patch release the VTC
bridge ownership earlier in unbind function instead of remove function
to avoid bridge ownership warning issue.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index d3a8e6885fc1..75cafb5b3035 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -3193,6 +3193,9 @@ void zynqmp_disp_unbind(struct device *dev, struct device *master, void *data)
 	struct zynqmp_dpsub *dpsub = dev_get_drvdata(dev);
 	struct zynqmp_disp *disp = dpsub->disp;
 
+	if (disp->vtc_bridge)
+		of_xlnx_bridge_put(disp->vtc_bridge);
+
 	zynqmp_disp_destroy_crtc(disp);
 	zynqmp_disp_destroy_plane(disp);
 	drm_property_destroy(disp->drm, disp->bg_c2_prop);
@@ -3407,8 +3410,6 @@ int zynqmp_disp_remove(struct platform_device *pdev)
 	zynqmp_disp_layer_destroy(disp);
 	if (disp->audclk)
 		zynqmp_disp_clk_disable(disp->audclk, &disp->audclk_en);
-	if (disp->vtc_bridge)
-		of_xlnx_bridge_put(disp->vtc_bridge);
 	zynqmp_disp_clk_disable(disp->aclk, &disp->aclk_en);
 	zynqmp_disp_clk_disable(disp->pclk, &disp->pclk_en);
 	dpsub->disp = NULL;
-- 
2.31.1

